### ISSUE TITLE
add* command speed ups

### ISSUE THREAD

**Issue by Matt529** [FRC-RS/FRS#19]
We have already optimized database queries and parallelized the data aggregation commands to some degree, but we could pursue further optimization two ways:

1. Forcing the work to be parallel, with clever use of `select_for_update`, atomic transactions and so on. Having tested brute force (which worked fairly well) there are still issues, particularly with concurrent modification and, because of the nature of the code, having to use Thread-level task parallelism and not Process-level parallelism leaving us subject to GIL.

2. Re-designing the commands to be properly done in parallel, there is some indication that most if not all of the tasks could be done in parallel if we reduce database contact as much as possible until the very end. Currently, the code is not conducive to such things.

3. Other clever inter-lingual solutions requiring a lot of work.

Because this operation is done *so* infrequently, I am running some final optimization tests and will be finished unless we decide to proceed further. We have achieved great speed increases thus far but there is little to be done without a lot more work. 

This is also low priority since there are plans to hook into the TBA API to instantly get new data and if we add support for large swaths of data, there should be appropriate options to  specifically grab that data (say, for supporting 2009).
